CDCgov/prime-reportstream
# 4246 Add new endpoint to fetch submission history from a report ID

This PR adds a new endpoint to fetch submission history from a report ID.

Test Steps:
1. Run your local RS on this branch.
2. Submit a new report or identify an existing report ID in your database.  You can get one from the report_lineage table.  You will also need the organization related to that submission.
3. Make a GET request to `http://localhost:7071/api/history/<org>/report/<report_id>`
4. Verify you get a response.
5. Obtain the action ID for the same report and make a GET request to `http://localhost:7071/api/history/<org>/report/<action_id>`.
6. Verify the response is the same as before.

## Changes
- Added report endpoint to get history for a report
- Minor lint fixes
- Refactored Jackson mapper into a common location.  Work to be done is to try to use only one configuration of the mapper if it is possible.  Seems like we could, but requires testing.

## Checklist

### Testing

### Process

## Fixes
- Closes #4246

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
